libvast/src/system/datagram_source.cpp

```diff
@@ -105,8 +107,23 @@ caf::behavior datagram_source(
       caf::arraybuf<> buf{msg.buf.data(), msg.buf.size()};
       self->state.reader->reset(std::make_unique<std::istream>(&buf));
       auto push_slice = [&](table_slice slice) {
-        VAST_DEBUG("{} produced a slice with {} rows", self, slice.rows());
-        self->state.mgr->out().push(detail::framed{std::move(slice)});
+        const auto unfiltered_rows = slice.rows();
+        if (self->state.filter) {
+          if (auto filtered_slice
+              = filter(std::move(slice), *self->state.filter)) {
+            VAST_VERBOSE("{} forwards {}/{} produced {} events after filtering",
+                         self, filtered_slice->rows(), unfiltered_rows,
+                         slice.layout().name());
+            self->state.mgr->out().push(std::move(*filtered_slice));
+          } else {
+            VAST_VERBOSE("{} forwards 0/{} produced {} events after filtering",
+                         self, unfiltered_rows, slice.layout().name());
+          }
+        } else {
+          VAST_VERBOSE("{} forwards {} produced {} events", self,
+                       unfiltered_rows, slice.layout().name());
+          self->state.mgr->out().push(std::move(slice));
+        }
```

Comment:
The log severity for all of those messages has to be `DEBUG`.

Reply:
Done. I also moved the logic to a function in the source state that is used for both source types.